x/storage/keeper/permission.go

```diff
@@ -119,14 +120,71 @@ func (k Keeper) VerifyGroupPermission(ctx sdk.Context, groupInfo *types.GroupInf
 	}
 
 	// verify policy
-	effect := k.permKeeper.VerifyPolicy(ctx, groupInfo.Id, gnfdresource.RESOURCE_TYPE_GROUP, operator, action, nil)
+	effect := k.VerifyPolicy(ctx, groupInfo.Id, gnfdresource.RESOURCE_TYPE_GROUP, operator, action, nil)
 	if effect == permtypes.EFFECT_ALLOW {
 		return permtypes.EFFECT_ALLOW
 	}
 
 	return permtypes.EFFECT_DENY
 }
 
+func (k Keeper) VerifyPolicy(ctx sdk.Context, resourceID math.Uint, resourceType gnfdresource.ResourceType,
+	operator sdk.AccAddress, action permtypes.ActionType, opts *permtypes.VerifyOptions) permtypes.Effect {
+	// verify policy which grant permission to account
+	policy, found := k.permKeeper.GetPolicyForAccount(ctx, resourceID, resourceType, operator)
+	if found {
+		effect, newPolicy := policy.Eval(action, ctx.BlockTime(), opts)
+		k.Logger(ctx).Info(fmt.Sprintf("CreateObject LimitSize update: %s, effect: %s, ctx.TxBytes : %d",
+			newPolicy.String(), effect, ctx.TxSize()))
+		if effect != permtypes.EFFECT_UNSPECIFIED {
+			if effect == permtypes.EFFECT_ALLOW && action == permtypes.ACTION_CREATE_OBJECT && newPolicy != nil && ctx.TxBytes() != nil {
+				_, err := k.permKeeper.PutPolicy(ctx, newPolicy)
```

Comment:
the method name `VerifyPolicy` seems like a read-only method, but the `PutPolicy` here is a bit strange

Reply:
Modified only in the execution of the transaction. Or do you have a recommended name?